demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -0,0 +1,1045 @@
+                     * If the period has elapsed, send out MQTT PINGREQ to the broker.  */
+                    timeDiff = calculateElapsedTime( currentTimeStamp.tv_sec, lastControlPacketSentTimeStamp );
+                    LogInfo( ( "Time Since last control packet %u \r\n", timeDiff ) );
+
+                    if( timeDiff >= MQTT_KEEP_ALIVE_PERIOD_SECONDS )
+                    {
+                        /* Send PINGREQ to the broker */
+                        LogInfo( ( "Sending PINGREQ to the broker\n " ) );
+                        mqttKeepAlive( tcpSocket, &fixedBuffer );
+                        controlPacketSent = true;
+                    }
+
+                    /* Since PUBLISH packet is not sent for this iteration, set publishPacketSent to false
+                     * so the next iteration will send PUBLISH .*/
+                    publishPacketSent = false;
+                }
+
+                if( controlPacketSent == true )
+                {
+                    /* Reset the last control packet sent timestamp */
+                    status = clock_gettime( CLOCK_MONOTONIC, &currentTimeStamp );
+                    assert( status == 0 );
+                    lastControlPacketSentTimeStamp = currentTimeStamp.tv_sec;
+                    controlPacketSent = false;
+
+                    /* Since the application is subscribed publishing messages to the same topic,
+                     * the broker will send the same message back to the application.
+                     * Process incoming PUBLISH echo or PINGRESP. */
+                    mqttProcessIncomingPacket( tcpSocket, &fixedBuffer );
+                }
+
+                /* Sleep until keep alive time period, so that for the next iteration this
+                 * loop will send out a PINGREQ if PUBLISH was noy sent for this iteration.
+                 * The broker will wait till 1.5 times keep-alive period before it disconnects
+                 * the client. */
+                sleep( MQTT_KEEP_ALIVE_PERIOD_SECONDS );
+            }
+
+            /* Unsubscribe from the previously subscribed topic */
+            LogInfo( ( "Unsubscribe from the MQTT topic %s.\r\n", MQTT_EXAMPLE_TOPIC ) );
+            mqttUnsubscribeFromTopic( tcpSocket, &fixedBuffer );
+            /* Process Incoming packet from the broker. */
+            mqttProcessIncomingPacket( tcpSocket, &fixedBuffer );
+
+            /* Send an MQTT Disconnect packet over the already connected TCP socket.
+             * There is no corresponding response for the disconnect packet. After sending
+             * disconnect, client must close the network connection. */
+            LogInfo( ( "Disconnecting the MQTT connection with %s.\r\n", MQTT_EXAMPLE_TOPIC ) );
+            mqttDisconnect( tcpSocket, &fixedBuffer );
+
+            /* Close the TCP connection. */
+            shutdown( tcpSocket, SHUT_RDWR );
+            close( tcpSocket );
```

Comment:
Do we intend for the demo to be MISRA compliant (I noticed an empty else at 413)? If so, these need to be voided since they return `int`s

Reply:
Will try to keep as compliant as possible, so I'll fix these.